app/scripts/modules/core/src/domain/ITrigger.ts

```diff
@@ -27,3 +27,8 @@ export interface IPipelineTrigger extends ITrigger {
 export interface ICronTrigger extends ITrigger {
   cronExpression: string;
 }
+
+export interface IPubsubTrigger extends ITrigger {
+  pubsubType: string;
```

Comment:
This name doesn't feel great to me, since it conflicts in my mind with a trigger's `type`. But I don't have any better suggestions (maybe `system` or `pubsubSystem`)?